### PR TITLE
Use `get(..)` instead of [..] for safe lookup of value (Fixes #7240)

### LINKED ISSUE #7240: OnePassword library assumes incorrectly that a field als an attribute 'value'
### Summary

When using the 'lookup' method to get a field value from a field in a OnePassword Vault, the method might raise a KeyError when trying to access the key 'value' on the field dict. Given a template in the OnePassword vault, a label for a field might exist without a value if the user has omitted this value on purpose.

### Issue Type

Bug Report

### Component Name

onepassword

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.15.3]
  python version = 3.9.17 (main, Jun 20 2023, 17:20:08) [Clang 14.0.3 (clang-1403.0.22.14.1)] 
  jinja version = 3.0.1
```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general
community.general 7.3.0  
community.general 2.0.1  

```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### OS / Environment

_No response_

### Steps to Reproduce

```yaml (paste below)
1. Create a onepassword item
2. Add a field with a label 'some-omitted-property'
3. Leave the value empty
4. Add a rule lookup(..., 'some-omitted-property') to an ansible script
5. Run ansible
```

### Expected Results

Empty (or None-like) value at the location of lookup(..., 'some-omitted-property')

### Actual Results

```console (paste below)
KeyError("value", "value")
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by wkleinheerenbrink**
A OnePassword field item might not have a value (property) when the user has omitted it (on purpose).

##### ISSUE TYPE

- Bugfix Pull Request (fixes #7240 )

##### COMPONENT NAME

OnePassword

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

File: plugins/lookup/onepassword.py [hunk 1]
```diff
@@ -462,10 +462,10 @@ def _parse_field(self, data_json, field_name, section_title=None):
                 # If the field name doesn't exist in the section, match on the value of "label"
                 # then "id" and return "value"
                 if field.get("label") == field_name:
-                    return field["value"]
+                    return field.get("value")
```
Comment (samdoran) [inline plugins/lookup/onepassword.py:465]: Use `""` for the default value.
```suggestion
                    return field.get("value", "")
```

**Comment by samdoran** [comment]
It would be good to add tests for this. There is already a test written, you just need to provide new fixture data.

A new file with the output from `op item get [item name] --format json` (minus any sensitive data) needs to go in `tests/unit/plugins/lookup/onepassword_fixtures/v2_out_04.json`.

A new entry in the test matrix needs to go [here](https://github.com/ansible-collections/community.general/blob/bc1df329a21b2a8602e7f59928360b60005a9eff/tests/unit/plugins/lookup/onepassword_common.py#L83). It will look something like this:

```python
{
    # Request data from a custom field with no value
    "vault_name": "Test Vault",
    "queries": ["Dummy Login"],
    "kwargs": {
        "field": "some-omitted-property",
    },
    "expected": [""],
    "output": load_file("v2_out_04.json")
},
```

You can run the test from the root of the collection by running `ansible-test units --docker default tests/unit/plugins/lookup/test_onepassword.py --python 3.11`.

That's a lot so please feel free to ask for help. Thanks for fixing this!

**Comment by samdoran** [comment]
Looks great!

shipit

**Comment by felixfontein** [comment]
@wkleinheerenbrink thanks for your contribution!
@samdoran thanks for reviewing!

**Pull request merged**